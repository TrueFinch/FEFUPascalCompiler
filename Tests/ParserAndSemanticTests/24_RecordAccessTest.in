program Project1;

type 
    a = record
        k: integer;
        l: integer;
    end;
  
var 
    i: a;
    r: integer;
begin
    r := i.k;
    writeln(r);
    r := i.l;
    writeln(r);
end.