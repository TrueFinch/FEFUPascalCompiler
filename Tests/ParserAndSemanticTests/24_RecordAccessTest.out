                                                  project1                                                    
                                                      ┼                                                       
                                                  MainBlock                                                   
             ┌───────────────────────────┬────────────┴───────────────────────────┐                           
           type                         var                               CompoundStatement                   
             ┼                    ┌──────┴───────┐               ┌────────────┬───┴────────┬────────────┐     
         TypeDecl           SimpleVarDecl  SimpleVarDecl        :=         writeln        :=         writeln  
 ┌───────────┴┐             ┌──┬──┴        ┌─────┼        ┌──────┴┐        ┌──┴     ┌──────┴┐        ┌──┴     
 a        RecordType        i  a           r  integer     r  RecordAccess  r        r  RecordAccess  r        
        ┌─────┴─────┐                                        ┌──┬─┴                    ┌──┬─┴                 
        :           :                                        i  k                      i  l                   
    ┌───┴─┐     ┌───┴─┐                                                                                       
    k  integer  l  integer                                                                                    
